src/classes/CacheProvider.js

```diff
@@ -0,0 +1,17 @@
+import { default as BaseProvider } from './BaseProvider';
+
+/**
+* Abstract class the provides a reference to an adapter instance (a caching implementation)
+*
+* @class
+* @extends {BaseProvider}
+* @param {Object} adapter - A cache adapter
+*/
+export class CacheProvider extends BaseProvider {
+  constructor() {
+    super(...arguments);
+    this.DEFAULT_ADAPTER = './MemoryCache';
```

Comment:
I would remove this DEFAULT_ADAPTER thing as is couples adapters and providers.

A better design is maybe to do something like that in the ParseServer init:

``` 
AProvider.setup(options.aprovider, aProviderDefaults)
```

That will enhance readability and debugging as all initialization would be done at the same place in the init.

Then the BaseProvider could implement the defaults/options loading mechanism.
The subclasses perform further initialization on the options.